pkg/nuclide-rust/lib/BuckIntegration.js

```diff
@@ -0,0 +1,137 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the license found in the LICENSE file in
+ * the root directory of this source tree.
+ *
+ * @flow
+ * @format
+ */
+
+import type {BusySignalService, BusySignalOptions} from 'atom-ide-ui';
+import type {TaskInfo} from '../../nuclide-buck/lib/types';
+import type {
+  AtomLanguageService,
+  LanguageService,
+} from '../../nuclide-language-service';
+
+import {getLogger} from 'log4js';
+import fsPromise from 'nuclide-commons/fsPromise';
+import {
+  getRustInputs,
+  getSaveAnalysisTargets,
+  normalizeNameForBuckQuery,
+  getRustBuildFile,
+} from './BuckUtils';
+
+import * as BuckService from '../../nuclide-buck-rpc';
+
+const logger = getLogger('nuclide-rust');
+
+export async function updateRlsBuildForTask(
+  task: TaskInfo,
+  service: AtomLanguageService<LanguageService>,
+  busySignalService: ?BusySignalService,
+) {
+  const buildTarget = normalizeNameForBuckQuery(task.buildTarget);
+
+  // Output is relative to Buck root but the built target may be managed by a
+  // Buck cell (nested Buck root).
+  // Here, Buck returns input paths relative to the possible cell, but the build
+  // file always relative to the current Buck root. Because of that, we use the
+  // build file path to determine the possible Buck cell root to which the
+  // inputs are relative to.
+  // FIXME: This is a bug in Buck, only query for files when the output is fixed.
+  const [relativeBuildFile, files] = await Promise.all([
```

Comment:
Before issuing these 2 potentially-expensive buck calls, could we verify that's actually a Rust target? (maybe using `task.buildRuleType.type`?

Reply:
@mostafaeweda my bad, I misunderstood what you were saying - I thought the gist was there being 2 calls and not checking before issuing them. Sorry! Will fix that in a moment.

Reply:
Done [here](https://github.com/facebook/nuclide/pull/1606/commits/7b9c326d1646262b6270e2e5c94353b96c83d152#diff-d7cc36e67cb07d1c82a31ad33a2544efR38).